jina/executors/indexers/keyvalue.py

```diff
@@ -104,7 +104,7 @@ def delete(self, keys: Iterator[int], *args, **kwargs):
             if self.query_handler:
                 self.query_handler.header[key] = None
             self._size -= 1
-
+        self.touch()
```

Comment:
it is weird the need for touch here. Shouldn't the update be handled automatically?

Reply:
true, it should not be the responsibility of the derived classes to handle this. I'll try to find a way around.

Reply:
Now, I added the functions `update` and `delete` to `update_funcs` in the `def register_class(cls):`  function in the `ExecutorType`  class.
Therefore, the `as_update_method` decorator is applied to the `update` and `delete` function.
Thanks for making your point.